pkg/olm/catalogsource_test.go

```diff
@@ -0,0 +1,327 @@
+			client:              true,
+		},
+		{
+			name:                "catalogsource",
+			namespace:           "test-namespace",
+			addToRuntimeObjects: false,
+			expectedError:       fmt.Errorf("catalogsource object catalogsource does not exist in namespace test-namespace"),
+			client:              true,
+		},
+		{
+			name:                "catalogsource",
+			namespace:           "test-namespace",
+			addToRuntimeObjects: true,
+			expectedError:       fmt.Errorf("catalogsource 'apiClient' cannot be empty"),
+			client:              false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		// Pre-populate the runtime objects
+		var runtimeObjects []runtime.Object
+
+		var testSettings *clients.Settings
+
+		catalogSource := generateCatalogSource(testCase.name, testCase.namespace)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, catalogSource)
+		}
+
+		if testCase.client {
+			testSettings = clients.GetTestClients(clients.TestClientParams{
+				K8sMockObjects: runtimeObjects, SchemeAttachers: testSchemes})
+		}
+
+		builderResult, err := PullCatalogSource(testSettings, testCase.name, testCase.namespace)
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, testCase.name, builderResult.Object.Name)
+			assert.Equal(t, testCase.namespace, builderResult.Object.Namespace)
+		}
+	}
+}
+
+func TestCatalogSourceGet(t *testing.T) {
+	testCases := []struct {
+		catalogSource *CatalogSourceBuilder
+		expectedError error
+	}{
+		{
+			catalogSource: buildValidCatalogSourceBuilder(buildTestClientWithDummyObject()),
+			expectedError: nil,
+		},
+		{
+			catalogSource: buildInValidCatalogSourceBuilder(buildTestClientWithDummyObject()),
+			expectedError: fmt.Errorf("catalogsource 'nsname' cannot be empty"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		catalogSource, err := testCase.catalogSource.Get()
+		assert.Equal(t, err, testCase.expectedError)
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, catalogSource.Name, testCase.catalogSource.Definition.Name)
+		}
+	}
+}
+
+func TestCatalogSourceExist(t *testing.T) {
+	testCases := []struct {
+		catalogSource  *CatalogSourceBuilder
+		expectedStatus bool
+	}{
+		{
+			catalogSource:  buildValidCatalogSourceBuilder(buildTestClientWithDummyObject()),
+			expectedStatus: true,
+		},
+		{
+			catalogSource:  buildInValidCatalogSourceBuilder(buildTestClientWithDummyObject()),
+			expectedStatus: false,
+		},
+		{
+			catalogSource:  buildValidCatalogSourceBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedStatus: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		exist := testCase.catalogSource.Exists()
+		assert.Equal(t, exist, testCase.expectedStatus)
+	}
+}
+
+func TestCatalogSourceCreate(t *testing.T) {
+	testCases := []struct {
+		catalogSource *CatalogSourceBuilder
+		expectedError error
+	}{
+		{
```

Comment:
Again, I think it is more useful to test Create when the resource does and doesn't exist. Other scenarios are just testing different functions